models/repo.go

```diff
@@ -1451,7 +1451,14 @@ func GetRepositoryByID(id int64) (*Repository, error) {
 
 // GetUserRepositories returns a list of repositories of given user.
 func GetUserRepositories(userID int64, private bool, page, pageSize int) ([]*Repository, error) {
-	sess := x.Where("owner_id = ?", userID).Desc("updated_unix")
+	sess := x.Where("owner_id = ?", userID).Desc("uid").Desc("updated_unix")
+
+	if setting.UsePostgreSQL {
+		sess = sess.Join("LEFT", "star", `"repository".id=star.repo_id AND star.uid = ?`, userID)
```

Comment:
I don't think that it's best practice to add conditions based on the used database. XORM can make it abstract enough to being valid for all supported databases.